### PR TITLE
Multi file dump loading

### PR DESCRIPTION
**Pull request by eugene-shevchenko**
Adds functionality for loading blockchain from several dump files.

### CODE REVIEW

File: ethereumj-core/src/main/java/org/ethereum/Start.java [hunk 1]
```diff
@@ -17,45 +17,79 @@
...
+import static org.apache.commons.lang3.StringUtils.isEmpty;
 
 /**
  * @author Roman Mandeleil
  * @since 14.11.2014
  */
 public class Start {
 
-    public static void main(String args[]) throws IOException, URISyntaxException {
+    private static void disableSync(SystemProperties config) {
```
Comment (mkalinin) [inline ethereumj-core/src/main/java/org/ethereum/Start.java]: Let's move helper methods to the end of class definitions.

File: ethereumj-core/src/main/java/org/ethereum/manager/BlockLoader.java [hunk 2]
```diff
@@ -18,136 +18,196 @@
...
+                }
+
+                @Override
+                public byte[] next() {
+                    return Hex.decode(scanner.nextLine());
+                }
+            };
+        }
+    }
+}
```
Comment (mkalinin) [inline ethereumj-core/src/main/java/org/ethereum/manager/BlockLoader.java]: new line

File: ethereumj-core/src/test/java/org/ethereum/manager/BlockLoaderTest.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    @Test
+    public void test() throws IOException {
+        Path[] paths = Arrays.stream(resource.getFile().listFiles())
+                .sorted()
+                .map(dmp -> Paths.get(dmp.toURI()))
+                .toArray(Path[]::new);
+
+        Assert.assertTrue(blockLoader.loadBlocks(paths));
+    }
+}
```
Comment (mkalinin) [inline ethereumj-core/src/test/java/org/ethereum/manager/BlockLoaderTest.java]: new line

File: ethereumj-core/src/main/java/org/ethereum/manager/BlockLoader.java [hunk 4]
```diff
@@ -18,136 +18,196 @@
...
+        initPipelines();
 
-        String fileSrc = config.blocksLoader();
+        AtomicLong maxBlockNumber = new AtomicLong();
+        boolean allBlocksImported;
         try {
-            final String blocksFormat = config.getConfig().hasPath("blocks.format") ? config.getConfig().getString("blocks.format") : null;
-            System.out.println("Loading blocks: " + fileSrc + ", format: " + blocksFormat);
-
-            if ("rlp".equalsIgnoreCase(blocksFormat)) {     // rlp encoded bytes
```
Comment (zilm13) [inline ethereumj-core/src/main/java/org/ethereum/manager/BlockLoader.java]: We were working with 2 different formats of block dumps: 
1) Our own format, dump is fired by `record.blocks` parameter in config
2) Hive format, `rlp`

Maybe I'm missing something but I see only one possible format parsing in new code

File: ethereumj-core/src/main/java/org/ethereum/Start.java [hunk 5]
```diff
@@ -17,45 +17,79 @@
...
 
         final SystemProperties config = SystemProperties.getDefault();
-        final boolean actionBlocksLoader = !config.blocksLoader().isEmpty();
-        final boolean actionGenerateDag = !StringUtils.isEmpty(System.getProperty("ethash.blockNumber"));
 
-        if (actionBlocksLoader || actionGenerateDag) {
-            config.setSyncEnabled(false);
-            config.setDiscoveryEnabled(false);
-        }
+        getEthashBlockNumber().ifPresent(blockNumber -> {
```
Comment (zilm13) [inline ethereumj-core/src/main/java/org/ethereum/Start.java]: Could you detach this block of code to private method with Javadoc?

File: ethereumj-core/src/main/java/org/ethereum/Start.java [hunk 6]
```diff
@@ -17,45 +17,79 @@
...
             System.exit(0);
-        } else {
-            Ethereum ethereum = EthereumFactory.createEthereum();
+        });
+
+        Ethereum ethereum = EthereumFactory.createEthereum();
 
-            if (actionBlocksLoader) {
-                ethereum.getBlockLoader().loadBlocks();
+        getBlocksDumpPath(config).ifPresent(path -> {
```
Comment (zilm13) [inline ethereumj-core/src/main/java/org/ethereum/Start.java]: Could you detach this block of code to private method with Javadoc?

File: ethereumj-core/src/main/java/org/ethereum/manager/BlockLoader.java (hunk 4 above)
Comment (eugene-shevchenko) [reply ethereumj-core/src/main/java/org/ethereum/manager/BlockLoader.java]: Yep, there were 2 dump formats. RLP is hive specific format, so I removed its support from ethereumj. Now ethereumj supports only own dump format. But if you need another one, you could implement DumpWalker interface and pass it to loadBlocks method for specific dump format reading.

File: ethereumj-core/src/main/java/org/ethereum/Start.java (hunk 5 above)
Comment (eugene-shevchenko) [reply ethereumj-core/src/main/java/org/ethereum/Start.java]: Why do we need add more methods to this pretty simple logic? I guess the main method is readable and straightforward enough.

File: ethereumj-core/src/main/java/org/ethereum/manager/BlockLoader.java (hunk 4 above)
Comment (zilm13) [reply ethereumj-core/src/main/java/org/ethereum/manager/BlockLoader.java]: Ahhh I got it, cool!

File: ethereumj-core/src/main/java/org/ethereum/Start.java (hunk 5 above)
Comment (zilm13) [reply ethereumj-core/src/main/java/org/ethereum/Start.java]: I'd like to see it with only line `createEthereum` 😃

**Pull request merged**